### PR TITLE
Feathers backend

### PR DESCRIPTION
**Pull request by frinzekt**
Contains:
- endpoint for dogs with mongoose model validation
- auto-creation of first user on start up of the server (this is based on environment variables that can be changed for production)
- local strategy authentication JWT

Features to be completed in another pull request
- vuex

### CODE REVIEW

**Comment by frinzekt** [comment]
There has been a decision to shift the imageLinking towards a string that can be sent. This is because there is too much effort to setup a proper file storage.

This means that the task for uploading image in the platform itself will not be there.
The idea is that when filling up the form to upload:
1. A button called "How to upload" is present
2. When the button is clicked, a modal appears. The modal has a button upload to google drive which opens google drive in another tab. Furthermore, the modal shows the instructions of:
     1. Upload the image
     2. Right click the file
     3. press the text called "Get Link"
     4. Paste the image in the field

**Comment by frinzekt** [comment]
instructions to get the google creds are here
https://www.youtube.com/watch?v=-rcRf7yswfM

**Review by noobling — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+services:
+  db:
+    image: mongo
+    container_name: bfa_mongo
+    restart: unless-stopped
+    env_file:
+      - .env
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: admin
+      MONGO_INITDB_ROOT_PASSWORD: admin
```
Comment (noobling) [inline docker-compose.yml:12]: i assume you will overwrite this in production?

Comment (frinzekt) [reply docker-compose.yml:12]: in production, I won't be using a docker mongo. I'll shift it to maybe Atlas or something

**Pull request merged**